addons/hr_expense/tests/test_expenses.py

```diff
@@ -410,3 +410,20 @@ def test_reset_move_to_draft(self):
 
         self.assertEqual(expense.state, 'done', 'Expense state must be done after payment')
         self.assertEqual(expense_sheet.state, 'done', 'Sheet state must be done after payment')
+
+    def test_expense_from_attachments(self):
+        self.env.user.employee_id = self.expense_employee.id
+        attachment = self.env['ir.attachment'].create({
+            'datas': b"R0lGODdhAQABAIAAAP///////ywAAAAAAQABAAACAkQBADs=",
+            'name': 'file.png',
+            'res_model': 'hr.expense',
+        })
+        product = self.env['product.product'].search([('can_be_expensed', '=', True)])
+        if product:
+            product = product.filtered(lambda p: p.default_code == "EXP_GEN") or product[0]
```

Comment:
```suggestion
        product = self.env['product.product'].search([('can_be_expensed', '=', True)], limit=1)
```
I don't think there are any products with code `EXP_GEN` :wink: